packages/ui/src/components/Stepper/index.tsx

```diff
@@ -1,17 +1,85 @@
 import { css, keyframes } from '@emotion/react'
 import styled from '@emotion/styled'
 import type { ReactNode } from 'react'
-import { Children, Fragment } from 'react'
+import {
+  Children,
+  Fragment,
+  createContext,
+  isValidElement,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
 import { Bullet } from '../Bullet'
 import { Text } from '../Text'
 
-type Temporal = 'previous' | 'next' | 'current'
-
 const LINE_HEIGHT_SIZES = {
   small: 2,
   medium: 4,
 } as const
 
+type Temporal = 'previous' | 'next' | 'current'
+
+type ContextType = {
+  step: number
+  setStep: React.Dispatch<React.SetStateAction<number>>
+  interactive: boolean
+  size: 'medium' | 'small'
+}
+
+type StepperProps = {
+  animated?: boolean
+  /**
+   * When true, the user can navigate through the steps by clicking on the bullets
+   */
+  interactive?: boolean
+  /**
+   * Number of the active step
+   */
+  selected?: number
+  children: ReactNode[]
+  className?: string
+  labelPosition?: 'bottom' | 'right'
+  size?: 'small' | 'medium'
+  'data-testid'?: string
+  separator?: boolean
+}
+
+type StepProps = {
+  onClick?: (index: number) => void
+  /**
+   * Automatically attribued by the parent Stepper
+   */
+  index?: number
+  /**
+   * Whether the step is disabled
+   */
+  disabled?: boolean
+  /**
+   * Title of the step
+   */
+  title?: ReactNode
+  /**
+   * @deprecated
+   * Do not use with Stepper.Step: use prop "title" instead
+   */
+  children?: ReactNode
+}
+
+type StepperContentProps = {
+  children: ReactNode
+  animated: boolean
+  separator: boolean
+}
+
+const State = createContext<ContextType>({
```

Comment:
You can also externalize `createContext` into an individual file that you could call `StepperProvider.tsx` even if there is not much into it, it will increase readability